src/plugins/vis_builder/public/visualizations/table/to_expression.ts

```diff
@@ -0,0 +1,165 @@
+      params,
+      label,
+      aggType: agg.type.name,
+    };
+  };
+
+  let cnt = 0;
+  const schemas: any = {
+    metric: [],
+  };
+
+  if (!aggConfigs) {
+    return schemas;
+  }
+
+  const responseAggs = aggConfigs.getResponseAggs().filter((agg: IAggConfig) => agg.enabled);
+  const metrics = responseAggs.filter((agg: IAggConfig) => agg.type.type === 'metrics');
+  const isHierarchical = showMetricsAtAllLevels;
+  responseAggs.forEach((agg) => {
+    let skipMetrics = false;
+    const schemaName = agg.schema;
+
+    if (!schemaName) {
+      cnt++;
+      return;
+    }
+
+    if (schemaName === 'split_row' || schemaName === 'split_column') {
+      skipMetrics = responseAggs.length - metrics.length > 1;
+    }
+
+    if (!schemas[schemaName]) {
+      schemas[schemaName] = [];
+    }
+
+    if (!isHierarchical || agg.type.type !== 'metrics') {
+      schemas[schemaName]!.push(createSchemaConfig(cnt++, agg));
+    }
+    if (isHierarchical && (agg.type.type !== 'metrics' || metrics.length === responseAggs.length)) {
+      metrics.forEach((metric: any) => {
+        const schemaConfig = createSchemaConfig(cnt++, metric);
+        if (!skipMetrics) {
+          schemas.metric.push(schemaConfig);
+        }
+      });
+    }
+  });
+
+  return schemas;
+};
+
+export interface TableRootState extends RootState {
+  style: TableOptionsDefaults;
+}
+
+export const toExpression = async ({ style: styleState, visualization }: TableRootState) => {
+  const { aggConfigs, indexPattern, expressionFns } = await getAggExpressionFunctions(
+    visualization
+  );
+  const { id: indexId = '' } = indexPattern;
+  let [opensearchaggs] = expressionFns;
+
+  const {
+    perPage,
+    showPartialRows,
+    showMetricsAtAllLevels,
+    showTotal,
+    totalFunc,
+    percentageCol,
+  } = styleState;
+
+  const schemas = getVisSchemas(aggConfigs, showPartialRows, showMetricsAtAllLevels);
+
+  const metrics =
+    schemas.bucket && showPartialRows && !showMetricsAtAllLevels
+      ? schemas.metric.slice(-1 * (schemas.metric.length / schemas.bucket.length))
+      : schemas.metric;
+
+  const tableData = {
+    metrics,
+    buckets: schemas.bucket || [],
+    splitRow: schemas.split_row,
+    splitColumn: schemas.split_column,
+  };
+
+  const tableConfig = {
+    perPage,
+    percentageCol,
+    showPartialRows,
+    showMetricsAtAllLevels,
+    showTotal,
+    totalFunc,
+  };
+
+  const visConfig = {
+    ...tableConfig,
+    ...tableData,
+  };
+
+  // Update buildExpressionFunction to correctly handle optional arguments
+  const tableVis = buildExpressionFunction<TableVisExpressionFunctionDefinition>(
+    'opensearch_dashboards_table',
+    {
+      visConfig: JSON.stringify(visConfig),
+    }
+  );
+
+  opensearchaggs = buildExpressionFunction<OpenSearchaggsExpressionFunctionDefinition>(
+    'opensearchaggs',
+    {
+      index: indexId,
+      metricsAtAllLevels: showMetricsAtAllLevels,
+      partialRows: showPartialRows,
```

Comment:
they are not missing. but metricsAtAllLevels needs to be consistent with showMetricsAtAllLevels to make it function correctly.

Reply:
I agree, which is why i suggest updating the `getAggExpressionFunctions` function instead of redoing the same thing for table vis separately. When i initially created the `getAggExpressionFunctions` function, all the visualizations i was referring to did not use this value, but passing this value from `styleState` is totally fine

Reply:
Okay. I got it. I can change the helper func and update other visualizations.